### PR TITLE
3.2.4 `createRequire` to allow copy loader from esbuild to work in SEA

### PR DESCRIPTION
**Pull request by jackyzha0**
The JS Bindings that NAPI-RS produces use raw require calls for local files.

With esbuild, this is normally fine as you can use the copy loader to just move the .node addon to the right place but SEAs dont allow requiring from the fs: [nodejs.org/docs/latest-v20.x/api/single-executable-applications.html#requireid-in-the-injected-main-script-is-not-file-based](https://nodejs.org/docs/latest-v20.x/api/single-executable-applications.html#requireid-in-the-injected-main-script-is-not-file-based)

They recommend using createRequire instead

This change should allow native addons created by NAPI-RS to be used in SEAs

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 const { existsSync, readFileSync } = require('fs')
 const { join } = require('path')
+const { createRequire } = require('node:module')
+require = createRequire(__filename)
```
Comment (masad-frost) [inline index.js:10]: Why not const require make it local?

**Review by masad-frost — APPROVED** [review]
Do we have to list ruspty as external to the esbuild bundle now?

File: index.js (hunk 1 above)
Comment (jackyzha0) [reply index.js:10]: 🤷 the node docs recommend it like this, i assume theres some static analysis things at work here

**Comment by jackyzha0** [comment]
> Do we have to list ruspty as external to the esbuild bundle now?

no it should still bundle ok

**Pull request merged**